src/main/java/com/bookshop/controller/AuthController.java

```diff
@@ -0,0 +1,135 @@
+package com.bookshop.controller;
+
+import com.bookshop.dto.user.UserLoginRequestDto;
+import com.bookshop.dto.user.UserLoginResponseDto;
+import com.bookshop.dto.user.UserRegistrationRequestDto;
+import com.bookshop.dto.user.UserRegistrationResponseDto;
+import com.bookshop.exception.RegistrationException;
+import com.bookshop.security.AuthenticationService;
+import com.bookshop.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.ExampleObject;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("/auth")
+@RequiredArgsConstructor
+@RestController
+public class AuthController {
+    private static final String CODE_200 = "200";
+    private static final String TOKEN_GENERATED_DESCRIPTION =
+            "JWT token was generated successfully";
+    private static final String REGISTRATION_SUCCESSFUL =
+            "registration was successful";
+    private static final String CODE_204 = "204";
+    private static final String DELETED_USER_DESCRIPTION = "User deleted successfully";
+    private static final String ROLE_SET = "role was set to the user";
+    private static final String CODE_401 = "401";
+    private static final String CODE_401_DESCRIPTION =
+            "User should be authenticated to do that operation";
+    private static final String CODE_403 = "403";
+    private static final String CODE_403_DESCRIPTION =
+            "Only users with role \"ADMIN\" can do such operation";
+    private static final String CODE_500 = "500";
+    private static final String CODE_500_DESCRIPTION = "Internal server error";
+    private static final String INCORRECT_DATA =
+            """
+                {
+                    "timestamp": "*time of event*",
+                    "status": "INTERNAL_SERVER_ERROR",
+                    "errors": [
+                        "Incorrect user data"
+                    ]
+                }
+            """;
```

Comment:
Do you really need all this code? Remember "KISS"

Reply:
I thought it will look simple and nice. Codes, indeed, more for esthetic purpose, but example of exceptions are used in swagger, so I thought is useful for someone, who will read documentation. But you are not the first one, who propose to remove it, maybe I really should.